test/unit/puppetlabs/services/master/master_service_test.clj

```diff
@@ -70,3 +69,56 @@
                   (test-path! "cert_inventory" "target/master-service-test/inventory.txt"))))))
         (finally
           (fs/delete-dir test-dir))))))
+
+; This ensures that a test failure will be logged for the version check
+; test if the master for some reason stops calling into the version check
+; library
+(def version-check-success
+  (atom false))
+
+(defn version-check-test-fn
+  [request-values update-server-url]
+  (is (= {:group-id "puppets" :artifact-id "yoda"} (:product-name request-values)))
+  (is (= "http://notarealurl/" update-server-url))
+  (swap! version-check-success not))
+
+; This exists to prevent the master from creating SSL files, and thus make the test run
+; faster
+(defn dummy-retrieve-cert-function
+  [_]
+  (println "Do nothing"))
+
+(deftest version-check-test
+  (testing "master calls into the dujour version check library using the correct values"
+    (with-redefs
+
+      [version-check/check-for-updates! version-check-test-fn
+       ; Redefine the CA functions to ensure that the master does not create
```

Comment:
I did this since it stops the master service from creating SSL files, which sped up the test a bit. I also figured it would reduce the amount of things the master is doing on initialization since all we care about in this test is whether or not it calls into the `check-for-updates!` function. Further, this eliminates the need to manage a test directory in the test.

I'm happy to remove this though, and manage a test directory in the above test.